transport.go

```diff
@@ -86,7 +89,11 @@ type transport struct {
 var _ tpt.Transport = &transport{}
 
 // NewTransport creates a new QUIC transport
-func NewTransport(key ic.PrivKey) (tpt.Transport, error) {
+func NewTransport(key ic.PrivKey, psk pnet.PSK) (tpt.Transport, error) {
+	if len(psk) > 0 {
+		log.Error("QUIC doesn't support private networks yet.")
```

Comment:
Note: when we actually implement this, let's derive a new shared secret by hashing `"quic-pnet" + PSK`.

Reply:
@Stebalien I've proposed to use a HKDF for that in https://github.com/libp2p/specs/pull/250